jenkins-slave.rb

```diff
@@ -1,92 +1,214 @@
+      JENKINS_URL=""
+      JENKINS_SECRET=""
+      JENKINS_PATH="/usr/bin:/bin:/usr/sbin:/sbin"
+
+      USAGE="Usage: $(basename "${0}") -u|--url <URL> -s|--secret <SECRET> [-p|--path <PATH>][-h|--help]"
+      HELP=$(cat <<- EOT
+      This script configures the launchctl configuration for the jenkins-slave service.
+
+      Options:
+
+        -u|--url <URL>          Required URL to the JNLP endpoint of the Jenkins slave.
+        -s|--secret <SECRET>    Required secret for the slave node to authenticate against the master.
+        -p|--path <PATH>        Optional path to set. Defaults to '/usr/bin:/bin:/usr/sbin:/sbin'.
+        -h|--help               Show this help.
+
+      Example:
+
+        jenkins-slave-configure --url http://your-jenkins/computer/node/slave-agent.jnlp --secret ******
+
+        jenkins-slave-configure --url http://your-jenkins/computer/node/slave-agent.jnlp --secret ****** \
+          --path '/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin'
+      EOT
+      )
+
+      print_help() {
+        echo "${USAGE}"
+        echo
+        echo "${HELP}"
+        echo
+      }
+
+      echo_err() {
+        echo "${1}" >&2
+      }
+
+      error() {
+        echo_err "Error: ${1}"
+      }
+
+      while (( "$#" )); do
+        case "${1}" in
+          -u|--url)
+          if [ -n "${2}" ] && [ "${2:0:1}" != "-" ]; then
+              JENKINS_URL="${2}"
+              shift 2
+          else
+              error "Argument for ${1} is missing"
+              echo_err "${USAGE}"
+              exit 1
+          fi
+          ;;
+        -s|--secret)
+          if [ -n "${2}" ] && [ "${2:0:1}" != "-" ]; then
+              JENKINS_SECRET="${2}"
+              shift 2
+          else
+              error "Argument for ${1} is missing"
+              echo_err "${USAGE}"
+              exit 1
+          fi
+          ;;
+        -p|--path)
+          if [ -n "${2}" ] && [ "${2:0:1}" != "-" ]; then
+              JENKINS_PATH="${2}"
+              shift 2
+          else
+              error "Argument for ${1} is missing"
+              echo_err "${USAGE}"
+              exit 1
+          fi
+          ;;
+        -h|--help)
+          print_help
+          exit 0
+          ;;
+        *)
+          error "Unsupported argument: $1!"
+          echo_err "${USAGE}"
+          ;;
+        esac
+      done
+
+      if [[ "${JENKINS_URL}" == "" ]]; then
+        error "Required argument --url not given!"
+        echo_err "${USAGE}"
+        exit 2
+      fi
+
+      if [[ "${JENKINS_SECRET}" == "" ]]; then
+        error "Required argument --secret not given!"
+        echo_err "${USAGE}"
+        exit 2
+      fi
+
+      sed -i '' "s|REPLACE_PATH|${JENKINS_PATH}|g" "${PLIST_FILE}"
+      sed -i '' "s|REPLACE_URL|${JENKINS_URL}|g" "${PLIST_FILE}"
+      sed -i '' "s|REPLACE_SECRET|${JENKINS_SECRET}|g" "${PLIST_FILE}"
+    EOS
+  end
+
+  def plist_name
+    "org.jenkins-ci." + name
```

Comment:
ping @ronaldtse should we use new name here?

Reply:
I'm not sure anymore if this was a good choice. The default is `"homebrew.mxcl.#{name}"` (see https://rubydoc.brew.sh/Formula.html#plist_name-instance_method). So it shouldbe rather "homebrew.mxcl.jenkins-slave" to comply with the convention of homebrew, isn't it?

Reply:
@phuonghuynh @ronaldtse What do you think about this?

Reply:
From @ronaldtse 

> Let’s revert this line, since it’s not managed by Jenkins 

so I reverted this line to use Ribose name.

Reply:
@Weltraumschaf https://github.com/mcxl[`mxcl`] is GitHub username of Homebrew's original author, and therefore should only be used for packages created directly by them otherwise it becomes misleading.

In any case, this naming practice originates from Apple's `plist` configuration, where the package is namespaced to a domain name in reverse (which itself comes from Java). Is the name of the formula problematic for you? We could potentially change it to `org.jenkins-ci.#{name}`.

Reply:
The name is not really problematic for me. Frankly, it's just a name. As a strong advocate of clean code and software craftmenship I always question things and look for improvements. So my intention was to use name which helps an administrator of such a build agent to easily find what he is looking for. 

I fully understand that you used your company domain as reverse FQDN because you made the formula. But the resulting service name listed by `launchctl list` is IMHO quite misleading. Because when I visit <http://ribose.com> I can't spot any hint about the service. So my intention of that change was to make clearer whereto look for further information, if one wonder what this service is. But now I think my choice of `org.jenkins-ci.#{name}` is also not a good name.

So I propose another one: `homebrew.riboseinc.jenkins-slave`

This follows the style of mxcl: 

- It clearly indicates that it comes from homebrew
- it clearly indicates your GitHub name where I can find source and contact the maintainers
- it contains the formula name I can use with `brew info ...` and such

What do you think?